include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -1051,6 +1051,19 @@ struct OrtApi {
    * Prefer a value of 0 if your CPU usage is very high.
    */
   ORT_API2_STATUS(SetGlobalSpinControl, _Inout_ OrtThreadingOptions* tp_options, int allow_spinning);
+
+  /**
+   * Add a pre-allocated initializer to a session. If a model contains an initializer with a name
+   * that is same as the name passed to this API call, ORT will use this initializer instance
+   * instead of deserializing one from the model file. This is useful when you want to share
+   * the same initializer across sessions.
+   * \param name name of the initializer
+   * \param val OrtValue containing the initializer. Lifetime of 'val' and the underlying initializer buffer must be
+   * managed by the user (created using the CreateTensorWithDataAsOrtValue API) and it must outlive the session object
+   * to which it is added.
+   */
```

Comment:
I think this can be supported in python as well as this is a  new “gap” when compared to the C API. Looks like it should be pretty easy to do so. Let me do it once this is merged.